time_analysis.py

```diff
@@ -0,0 +1,34 @@
+import sys
+
+try:
+    from scipy import stats as scipy_stats
+except ModuleNotFoundError:
+    print("Need scipy for statistical analysis", file=sys.stderr)
+    sys.exit(1)
+
+if len(sys.argv) < 3:
+    print("Need two filenames to compare", file=sys.stderr)
+    sys.exit(1)
+
+values = [[int(n) for n in open(file, "r").readlines()]
+          for file in sys.argv[1:3]]
+
+# Get the mean/variance
+stats = list(map(scipy_stats.describe, values))
+# Calculate both with and without the assumption of equal variance
+p_eq_var = scipy_stats.ttest_ind(*values, equal_var=True).pvalue
+p_uneq_var = scipy_stats.ttest_ind(*values, equal_var=False).pvalue
+
+print("Mean")
+for stat in stats:
+    print(f"{stat.mean:.3f}")
+print()
+print("Standard deviation")
+for stat in stats:
+    print(f"{stat.variance**(1/2):.3f}")
+print()
+# Normal p-values are p<0.05 or even p<0.01 for statistical significance.
+# Perhaps try a baseline where you compare the programs with themselves.
+print("P values (small values -> reject equal mean hypothesis):")
+print("Equal variance   Unequal variance")
+print(f"{p_eq_var:.3f}            {p_uneq_var:.3f}")
```

Comment:
Haven't run it by it looks good!

There are some things that might be useful that isn't recorded. Maybe store hope many instructions the program is and the like, so we can compare compiler optimizations aswell?

We might be a little bit more markup in the data file, maybe a really dumb json writer that just blurts out the content? But that doesn't have to go on straight away.

How does this work for games? Can we make it useful there somehow? Counting frames maybe? Or until the yield keyword?

Reply:
Yeah this currently only makes sense for non-games. But I think game performance should focus more on metrics like "average frame time" and the likes while this focuses solely on "how long does this program take to terminate" and relies on /many/ runs to compare runtimes.